packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -463,10 +467,10 @@ class _MergingListenable extends Listenable {
 class ValueNotifier<T> extends ChangeNotifier implements ValueListenable<T> {
   /// Creates a [ChangeNotifier] that wraps this value.
   ValueNotifier(this._value) {
-    if (kFlutterMemoryAllocationsEnabled) {
+    if (kFlutterMemoryAllocationsEnabled && MemoryAllocations.instance.hasListeners) {
```

Comment:
Why do these calls need to be guarded by `MemoryAllocations.instance.hasListeners`? It makes this a little more cumbersome to use and it is another thing that's easy to forget when using this API. If this guard is indeed necessary, please update the documentation on dispatchObjectEvent and maybe assert in that method that it is only called when there are listeners?

Reply:
It is not necessary, but I want to save on object creation.
I ended up moving object creation to MemoryAllocations